client/src/components/Slider/PageSlider.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { ReactNode, useEffect, useRef, useState } from 'react';
+import { SLIDER_POSITION } from 'src/constants/sliderPosition';
+import styled from 'styled-components/macro';
+
+interface PageSliderProps {
+  children: ReactNode;
+  animationCallback: any;
+  pagePosition: string;
+}
+
+export default function PageSlider({
+  children,
+  animationCallback,
+  pagePosition,
+}: PageSliderProps) {
+  const [position, setPosition] = useState('');
+  const sliderRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    const startAnimation = () => {
+      setPosition(pagePosition);
+    };
+
+    const onTransitionEnd = (e: any) => {
+      if (e.propertyName !== 'transform') return;
+      animationCallback();
+    };
+
+    if (sliderRef.current) {
+      sliderRef.current.addEventListener('transitionend', onTransitionEnd);
+      startAnimation();
+    }
+  }, [children]);
+
+  const getClassName = () => {
+    let className = `animatable`;
+    if (SLIDER_POSITION.TO_LEFT === pagePosition)
+      className += ' prepare-to-left';
+    if (SLIDER_POSITION.FROM_RIGHT === pagePosition)
+      className += ' prepare-from-rigth';
+    if (SLIDER_POSITION.TO_LEFT === position) className += ' to left';
+    if (SLIDER_POSITION.FROM_RIGHT === position) className += ' from right';
+    return className;
+  };
+
+  return (
+    <Container>
+      <div ref={sliderRef} className={getClassName()}>
+        {children}
+      </div>
+    </Container>
+  );
```

Comment:
클래시 명을 string으로 관리하다보니 나중에 휴먼 에러?가 생길 수도 있을 것 같은데 styled-components 의 props를 활용해보는 건 어떨까요? 그러면 Container는 아래와 같은 형태가 될테고

```suggestion
    <Container ref={sliderRef} pagePosition={pagePosition}>
      {children}
    </Container>
```